application/apps/indexer/processor/src/grabber.rs

```diff
@@ -12,9 +12,12 @@ pub trait GrabTrait {
     fn grab_content(&self, line_range: &LineRange) -> Result<GrabbedContent, GrabError>;
     fn inject_metadata(&mut self, metadata: GrabMetadata) -> Result<(), GrabError>;
     fn get_metadata(&self) -> Option<&GrabMetadata>;
+    fn drop_metadata(&mut self);
     fn associated_file(&self) -> PathBuf;
 }
 
+pub trait AsyncGrabTrait: GrabTrait + Sync + Send + std::fmt::Debug {}
```

Comment:
do we really need to add those traits with this commit? I thought you only need them for the cancellation stuff

Reply:
This trait related to async functionality, but not to cancelation.